### PR TITLE
Data component

### PR DESCRIPTION
**Pull request by luucvanderzee**
- Replaced `vgg-transform` by `vgg-data` component
- Reprojection of coordinate systems now handled with `reproject` transformation
- Added some documentation to core components (not finished yet)
- Created some 'concepts' that might need explanation in more depth (also not finished yet)
- Removed some small bugs, like incorrect prop types in some places

The reason that I haven't finished the core components (actually mainly the `vgg-section` and `vgg-glyph` )is because these might still change a lot after we look into the faceting/data ID/reusable scales stuff. 

The concepts can maybe better be adressed in a new PR at some other moment, otherwise this one will contain too much changes.

### CODE REVIEW

File: docs/graphs/plottitle.md [hunk 1]
```diff
@@ -14,21 +14,18 @@ Although similar to the Label Mark, the plot title component assumes there will
...
-### Coordinate props
-
-Prop      | Required | Type             | Default    |  Description                 | Unit(s)                |
-----------|----------|------------------|------------|------------------------------|------------------------|
-vjust     | false    | [String, Number] | center     | Horizontal position of title | Number between 0 and 1 |
-hjust     | false    | [String, Number] | t          | Vertical position of title   | Number between 0 and 1 |
-margin    | false    | Number           | 50         | Margin of text               | Screen pixel           |   
-
-
-The `vjust` and `hjust` props are used to position the title relative to a section or the overall graphic. It is centered at the top by default.
```
Comment (atepoorthuis) [inline docs/graphs/plottitle.md]: Why remove this explanation? I thought it works well to clarify the options available for `vjust` and `hjust`?

File: docs/transform/binning.md [hunk 2]
```diff
@@ -5,21 +5,24 @@ title: Binning
...
+    :transform="{ binning: { groupBy: 'apple', method: selected, numClasses: 5 } }"
+>
 ```
 
 # Binning
 
-Binning is used within the `<vgg-transform`> tag. Similar to the groupBy transformation, binning groups the data into numeric bins. These bins can be determined one of the inbuilt methods/algorithms or provided manually as an array of [lowerBound, upperBound] ranges.
+Similar to the groupBy transformation, binning groups the data into numeric bins.
+These bins can be determined one of the inbuilt methods/algorithms or provided manually
+as an array of `[lowerBound, upperBound]` ranges.
```
Comment (atepoorthuis) [inline docs/transform/binning.md]:
```suggestion
as an array of bin boundaries.
```

File: docs/graphs/plottitle.md (hunk 1 above)
Comment (luucvanderzee) [reply docs/graphs/plottitle.md]: Oops, I think I removed that by accident

Comment (luucvanderzee) [reply docs/graphs/plottitle.md]: Fixed it in commit 46f8021

**Pull request merged**